hummingbot/connector/exchange/bitmax/bitmax_api_order_book_data_source.py

```diff
@@ -0,0 +1,260 @@
+#!/usr/bin/env python
+import asyncio
+import logging
+import aiohttp
+import websockets
+import ujson
+
+from typing import Optional, List, Dict, Any, AsyncIterable
+from hummingbot.core.data_type.order_book import OrderBook
+from hummingbot.core.data_type.order_book_message import OrderBookMessage
+from hummingbot.core.data_type.order_book_tracker_data_source import OrderBookTrackerDataSource
+from hummingbot.core.utils.async_utils import safe_gather
+from hummingbot.logger import HummingbotLogger
+from hummingbot.connector.exchange.bitmax.bitmax_active_order_tracker import BitmaxActiveOrderTracker
+from hummingbot.connector.exchange.bitmax.bitmax_order_book import BitmaxOrderBook
+from hummingbot.connector.exchange.bitmax.bitmax_utils import convert_from_exchange_trading_pair, convert_to_exchange_trading_pair, ms_timestamp_to_s
+from hummingbot.connector.exchange.bitmax.bitmax_constants import EXCHANGE_NAME, REST_URL, WS_URL
+
+
+class BitmaxAPIOrderBookDataSource(OrderBookTrackerDataSource):
+    MAX_RETRIES = 20
+    MESSAGE_TIMEOUT = 30.0
+    SNAPSHOT_TIMEOUT = 10.0
+
+    _logger: Optional[HummingbotLogger] = None
+
+    @classmethod
+    def logger(cls) -> HummingbotLogger:
+        if cls._logger is None:
+            cls._logger = logging.getLogger(__name__)
+        return cls._logger
+
+    def __init__(self, trading_pairs: List[str] = None):
+        super().__init__(trading_pairs)
+        self._trading_pairs: List[str] = trading_pairs
+        self._snapshot_msg: Dict[str, any] = {}
+
+    @classmethod
+    async def get_last_traded_prices(cls, trading_pairs: List[str]) -> Dict[str, float]:
+        result = {}
+
+        async with aiohttp.ClientSession() as client:
+            resp = await client.get(f"{REST_URL}/ticker")
+            if resp.status != 200:
+                raise IOError(
+                    f"Error fetching last traded prices at {EXCHANGE_NAME}. "
+                    f"HTTP status is {resp.status}."
+                )
+
+            resp_json = await resp.json()
+            if resp_json.get("code") != 0:
+                raise IOError(
+                    f"Error fetching last traded prices at {EXCHANGE_NAME}. "
+                    f"Error is {resp_json.message}."
+                )
+
+            for item in resp_json.get("data"):
+                trading_pair: str = convert_from_exchange_trading_pair(item.get("symbol"))
+                if trading_pair not in trading_pairs:
+                    continue
+
+                # we had to calculate mid price
+                ask = float(item.get("ask")[0])
+                bid = float(item.get("bid")[0])
+                result[trading_pair] = (ask + bid) / 2
```

Comment:
Looking at BitMax API documentation, I figure that this does not return the last traded price, instead it returns the current midprice of the orderbook. I think you're looking for the `GET /api/pro/v1/trades` endpoint.

Reply:
Definitely better yes, my bad.
Kinda sucks that we can't do something like `/api/pro/v1/trades?symbol=BTMX/USDT.BTC/USDT` .. etc